### ISSUE TITLE
Failing to build packages with Cargo nightly

### ISSUE THREAD

**Issue by glasserc** [NixOS/nixpkgs#61192]
## Issue description

As of https://github.com/rust-lang/cargo/pull/6880, cargo now creates locks in `$CARGO_HOME/.cargo/.package-cache`. When trying to build some cargo packages using Rust nightly (from https://github.com/mozilla/nixpkgs-mozilla/), I've gotten failures such as https://gist.github.com/glasserc/ec065e3cf7706c13742dc16fb79a931b.

### Steps to reproduce

I hit this while trying to build Gecko using mozilla/nixpkgs-mozilla#175. With that patch applied, from within a checkout of Gecko, I ran: `nix-shell ../nixpkgs-mozilla/release.nix -A gecko.x86_64-linux.gcc  --command fish` (and got the output shown above).

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.40, NixOS, 19.03.172500.aade6ded796 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(ethan): `""`
 - channels(root): `"nixos-19.03.172500.aade6ded796"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by Atemu** [comment]
Is this still an issue?

**Comment by glasserc** [comment]
I haven't tried recently but it looks like it should be fixed by commit 33723434a888b7b9ce5a9b35dba2e77baf254109, per comments in https://github.com/NixOS/nixpkgs/pull/61194.
